lib/engine-addon.js

```diff
@@ -387,10 +387,76 @@ module.exports = {
         return mergeTrees([
           externalTree,
           engineJSTree,
-          engineCSSTree
+          compiledEngineCSSTree
         ].filter(Boolean), { overwrite: true });
       };
 
+      this.compileLazyEngineStyles = function compileLazyEngineStyles(vendorTree, externalTree) {
+        var vendorCSSTree = buildVendorCSSTree.call(this, vendorTree);
```

Comment:
@stefanpenner - Yes, they are `memoize`ed with a custom `memoize` util function already.  We can clean it up a bit and make sure they are cached in the same way in future refactors.